examples/mongocxx/inserted_id.cpp

```diff
@@ -0,0 +1,41 @@
+// Copyright 2016 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iostream>
+
+#include <bsoncxx/types.hpp>
+#include <mongocxx/client.hpp>
+#include <mongocxx/instance.hpp>
+
+int main(int, char**) {
+    mongocxx::instance inst{};
+    mongocxx::client conn{mongocxx::uri{}};
+
+    bsoncxx::builder::stream::document document{};
+
+    auto collection = conn["test"]["col"];
+    document << "test" << 1;
+
+    auto result = collection.insert_one(document.view());
+
+    if (result) {
```

Comment:
Personally, I'm an anti-deep-nesting sort of person, so I'd reverse the sense here:

if (!result) {
    std::cout << ...
    return EXIT_FAILURE; // presumably this indicates something very wrong
};

// No need for an else clause, since otherwise we have returned.